demos/demo30_Deconvolution2D.py

```diff
@@ -85,8 +85,8 @@
 Dt = kron(P, I)
 P = Ds+Dt # Precision matrix defined for the 2D problem
 
-# Then we define the Gaussian using GaussianPrec to avoid computing the (dense) covariance matrix.
-TP.prior = cuqi.distribution.GaussianPrec(mean = np.zeros(n), prec = P)
+# Then we define the Gaussian using Gaussian precision to avoid computing the (dense) covariance matrix.
+TP.prior = cuqi.distribution.Gaussian(mean = np.zeros(n), prec = P)
```

Comment:
I think this would be better with the correct geometry
```suggestion
TP.prior = cuqi.distribution.Gaussian(mean=np.zeros(n), prec=P, geometry=TP.model.domain_geometry)
```

Reply:
The testproblem should automatically determine the geometry needed here. So its not needed to add the geometry to the prior.

Reply:
It is just that the prior samples that are generated in the next lines do NOT have the correct geometry so are plotted as default.

Reply:
Ah because of the sample prior issue in tp. Ok lets do it for now then